src/Hosting/IntegrationTesting/src/Deployers/ApplicationDeployer.cs

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Internal;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.Hosting.IntegrationTesting
+{
+    /// <summary>
+    /// Abstract base class of all deployers with implementation of some of the common helpers.
+    /// </summary>
+    public abstract class ApplicationDeployer : IDisposable
+    {
+        public static readonly string DotnetCommandName = "dotnet";
+
+        private readonly Stopwatch _stopwatch = new Stopwatch();
+
+        private PublishedApplication _publishedApplication;
+
+        public ApplicationDeployer(DeploymentParameters deploymentParameters, ILoggerFactory loggerFactory)
+        {
+            DeploymentParameters = deploymentParameters;
+            LoggerFactory = loggerFactory;
+            Logger = LoggerFactory.CreateLogger(GetType().FullName);
+
+            ValidateParameters();
+        }
+
+        private void ValidateParameters()
+        {
+            if (DeploymentParameters.RuntimeFlavor == RuntimeFlavor.None && !string.IsNullOrEmpty(DeploymentParameters.TargetFramework))
```

Comment:
if (DeploymentParameters.ServerType == ServerType.None)
            {
                throw new ArgumentException($"Invalid ServerType '{DeploymentParameters.ServerType}'.");
            }

was in the aspnetcore version.